### PR TITLE
KAFKA-5797: Handle metadata not available in store registration

### PR DESCRIPTION
**Pull request by guozhangwang**
This is the backport of #3748 for trunk

### CODE REVIEW

**Comment by guozhangwang** [comment]
@dguy @mjsax @bbejeck for reviews.

**Comment by bbejeck** [comment]
retest this please

**Comment by dguy** [comment]
@guozhangwang there are checkstyle failures

**Review by dguy — APPROVED** [review]
Thanks @guozhangwang, LGTM

**Comment by guozhangwang** [comment]
Merged to 0.11.0 branch.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java [hunk 1]
```diff
@@ -125,42 +88,75 @@ public void register(final StateRestorer restorer) {
...
+                if (restorer.checkpoint() >= endOffset) {
                     restorer.setRestoredOffset(restorer.checkpoint());
-                } else if (restorer.offsetLimit() == 0 || endOffsets.get(topicPartition) == 0) {
+                    iter.remove();
+                } else if (restorer.offsetLimit() == 0 || endOffset == 0) {
                     restorer.setRestoredOffset(0);
-                } else {
-                    newTasksNeedingRestoration.put(topicPartition, restorer);
+                    iter.remove();
                 }
```
Comment (tedyu) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java:132]: Should the following be called for the else case ?
```
restorer.setEndingOffset(endOff
```

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java:132]: In 0.11.0 branch the code is diverged from trunk already, in this case we used `Map<TopicPartition, Long> endOffsets` instead.

**Pull request closed without merging**